execute/exectypes/costly_messages.go

```diff
@@ -364,4 +376,55 @@ func (c *CCIPMessageExecCostUSD18Calculator) computeExecutionCostUSD18(
 	return cost
 }
 
+// computeDataAvailabilityCostUSD18 computes the data availability cost of a message in USD18s.
+func computeDataAvailabilityCostUSD18(
+	dataAvailabilityFee *big.Int,
+	daConfig cciptypes.DataAvailabilityGasConfig,
+	message cciptypes.Message,
+) plugintypes.USD18 {
+	if dataAvailabilityFee == nil || dataAvailabilityFee.Cmp(big.NewInt(0)) == 0 {
+		return big.NewInt(0)
+	}
+
+	messageGas := calculateMessageMaxDAGas(message, daConfig)
+	return big.NewInt(0).Mul(messageGas, dataAvailabilityFee)
+}
+
+// calculateMessageMaxDAGas calculates the total DA gas needed for a CCIP message
+func calculateMessageMaxDAGas(
+	msg cciptypes.Message,
+	daConfig cciptypes.DataAvailabilityGasConfig,
+) *big.Int {
+	// Calculate token data length
+	var totalTokenDataLen int
+	for _, tokenAmount := range msg.TokenAmounts {
+		totalTokenDataLen += len(tokenAmount.SourcePoolAddress) +
```

Comment:
given we are already calcing DA gas in EVM specific manner, we can follow the fee quoter logic:
```
total_len = fixed_overhead + len(tokenAmount.ExtraData) + len(tokenAmount.DestExecData)
where
fixed_overhead = ANY_2_EVM_MESSAGE_FIXED_BYTES_PER_TOKEN = 32 * ((2 * 3) + 3);
see Internal.sol for breakdown
```